### PR TITLE
[fix-devhub-enabled-expiry] Fix user is_devhub_enabled to prevent expiration.

### PR DESCRIPTION
**Pull request by wlonk**
See https://trello.com/c/OYUy81kP/75-bug-sf-token-expires-causing-isdevhubenabled-to-return-an-incorrect-falsy-value

This is just a first step, but I want to get it up so others can try it and help shake out problems.

Curious Zelda

### CODE REVIEW

**Comment by wlonk** [comment]
@davisagli Lemme know if this matches what you described in the Trello card.

**Comment by jgerigmeyer** [comment]
[Bug: SF token expires, causing `is_devhub_enabled` to return an incorrect falsy value.](https://trello.com/c/OYUy81kP/75-bug-sf-token-expires-causing-isdevhubenabled-to-return-an-incorrect-falsy-value)

File: metashare/api/models.py [hunk 1]
```diff
@@ -134,20 +135,19 @@ def valid_token_for(self):
...
-        instance_url = self.salesforce_account.extra_data["instance_url"]
-        url = f"{instance_url}/services/data/v45.0/sobjects/ScratchOrgInfo"
-        resp = requests.get(url, headers={"Authorization": f"Bearer {token}"})
-        if resp.status_code == 200:
-            return True
-        if resp.status_code == 404:
+        client = get_devhub_api(devhub_username=self.sf_username)
+        try:
+            resp = client.restful("sobjects/ScratchOrgInfo")
+            if resp:
```
[the remark this answers is not in the document]
Comment (jgerigmeyer) [reply metashare/api/models.py]: I'm certainly open to cleanup or suggestions, but yes I did functionally test it. `restful` does not return a response with HTTP status, but just returns either a JSON object or `None` (or raises an error). I wasn't sure how generically we want to catch exceptions though...

See https://github.com/simple-salesforce/simple-salesforce/blob/master/simple_salesforce/api.py#L290-L311

[the remark this answers is not in the document]
Comment (davisagli) [reply metashare/api/models.py]: @jgerigmeyer oh right, I forgot `restful` takes care of the JSON parsing.

I'm guessing it's a 404 response in the case where dev hub isn't enabled, but I don't know if the response body is json or not. Does restful's call to _call_salesforce raise an exception in that case?

[the remark this answers is not in the document]
Comment (jgerigmeyer) [reply metashare/api/models.py]: Correct. Here is the error when DH is not enabled:

```
  File "/app/metashare/api/models.py", line 145, in is_devhub_enabled
    resp = client.restful("sobjects/ScratchOrgInfo")
  File "/usr/local/lib/python3.8/site-packages/simple_salesforce/api.py", line 304, in restful
    result = self._call_salesforce(method, url, name=path, params=params,
  File "/usr/local/lib/python3.8/site-packages/simple_salesforce/api.py", line 478, in _call_salesforce
    exception_handler(result, name=name)
  File "/usr/local/lib/python3.8/site-packages/simple_salesforce/util.py", line 61, in exception_handler
    raise exc_cls(result.url, result.status_code, name, response_content)
simple_salesforce.exceptions.SalesforceResourceNotFound: Resource sobjects/ScratchOrgInfo Not Found. Response content: [{'errorCode': 'NOT_FOUND', 'message': 'The requested resource does not exist'}]
```

[the remark this answers is not in the document]
Comment (davisagli) [reply metashare/api/models.py]: Got it, this looks okay then.

File: metashare/api/models.py [hunk 2]
```diff
@@ -140,12 +141,13 @@ def is_devhub_enabled(self):
             return None
```
Comment (wlonk) [inline metashare/api/models.py]: We might wanna change the return type of this function to a strict `Boolean` and make these `None`s `False`s instead. Does returning `null` in the serializer have value to you?

**Pull request merged**